File: vripper-server/src/main/java/tn/mnlr/vripper/jpa/domain/LogEvent.java
package tn.mnlr.vripper.jpa.domain;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@NoArgsConstructor
public class LogEvent {

  private Long id;
  private Type type;
  private Status status;

  @JsonSerialize(using = DateTimeSerializer.class)
  private LocalDateTime time;

  private String message;

  public LogEvent(Type type, Status status, LocalDateTime time, String message) {
    this.type = type;
    this.status = status;
    this.time = time;
    this.message = message;
  }

  public enum Type {
    POST,
    QUEUED,
    THANKS,
    METADATA,
    SCAN,
    DOWNLOAD,

    // Below are deprecated events
    METADATA_CACHE_MISS,
    QUEUED_CACHE_MISS
  }

  public enum Status {
    PENDING,
    PROCESSING,
    DONE,
    ERROR
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/entities/LogEvent.kt
package me.mnlr.vripper.entities

import java.time.LocalDateTime

data class LogEvent(
    val id: Long? = null,
    val type: Type,
    val status: Status,
    val time: LocalDateTime = LocalDateTime.now(),
    val message: String,
) {

    enum class Type(val stringValue: String) {
        POST("Post"), THREAD("Thread"), THANKS("Thanks"), METADATA("Metadata"), SCAN("Scan"), DOWNLOAD(
            "Download"
        )
    }

    enum class Status(val stringValue: String) {
        PENDING("Pending"), PROCESSING("Processing"), DONE("Done"), ERROR("Error")
    }
}